components/base/kinematicbase/ptgKinematics.go

```diff
@@ -228,26 +228,25 @@ func (ptgk *ptgBaseKinematics) goToInputs(ctx context.Context, inputs []referenc
 			defer cancelFn()
 			return multierr.Combine(err, ptgk.Base.Stop(stopCtx, nil))
 		}
-		utils.PanicCapturingGo(func() {
-			// We need to update currentInputs as we move through the arc.
-			for timeElapsed := 0.; timeElapsed <= step.timestepSeconds; timeElapsed += inputUpdateStep {
-				distIncVel := step.linVelMMps.Y
-				if distIncVel == 0 {
-					distIncVel = step.angVelDegps.Z
-				}
-				ptgk.inputLock.Lock()
-				ptgk.currentInput = []referenceframe.Input{inputs[0], inputs[1], {math.Abs(distIncVel) * timeElapsed}}
-				ptgk.inputLock.Unlock()
-				utils.SelectContextOrWait(ctx, time.Duration(inputUpdateStep*1000*1000)*time.Microsecond)
-			}
-		})
 
-		if !utils.SelectContextOrWait(ctx, timestep) {
-			ptgk.logger.CDebug(ctx, ctx.Err().Error())
-			// context cancelled
-			break
+		// We need to update currentInputs as we move through the arc.
+		for timeElapsed := 0.; timeElapsed <= step.timestepSeconds; timeElapsed += inputUpdateStep {
+			if ctx.Err() != nil {
+				ptgk.logger.CDebug(ctx, ctx.Err().Error())
+				// context cancelled
+				return ctx.Err()
```

Comment:
Ah; one thing that's missing here, we want to stop the base if this happens.

See above
```
	if err != nil {
		stopCtx, cancelFn := context.WithTimeout(context.Background(), time.Second*5)
		defer cancelFn()
		return multierr.Combine(err, ptgk.Base.Stop(stopCtx, nil))
	}
```

Reply:
ah, thank you for catching this :D

Reply:
getting `defer within an immediate outer for usage found "defer cancelFn()"` from linter. I will have to play around a little with this.